### PR TITLE
Tagging [approved with caveats]

### PR DESCRIPTION
**Pull request by jimrandomh**
Introduces tagging, gated admin-only. Tags are created by admins, then applied to posts; when a post has a tag, users can vote on its relevance. The footer of post-pages has a list of tags, and UI for adding more.

New collections: `tags` and `tagRels`. A `tagRel` represents the relationship between a post and a tag which has been applied to it. The `tags` collection is Algolia-indexed; the `tagRel` collection is voteable.

New pages:
* `/tag/tag-name`: Shows the tag's description, and a list of posts with that tag, sorted by relevance.
* `/tag/tag-name/edit`: Admin-only. Edit the description of a tag.
* `/tag/create`: Admin-only. Create a new tag.
* `/tags`: An index of all tags that exist. Not currently linked to from anywhere.

Todos:
- [x] Validation and edge cases in `addOrUpvoteTag`
- [x] Finish styling the post footer
- [x] Introduce `PostsItemPlaceholder` for the post footer hover
- [x] Make links to tag pages have a hover-preview
- [x] Add-tag popup should have a hover which shows the tag description
- [x] Debug some Algolia issues

### CODE REVIEW

**Comment by Raemon** [comment]
(note: I just sent an update fixing the merge conflicts)

File: packages/lesswrong/components/BenchmarkComponent.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Components, registerComponent } from 'meteor/vulcan:core';
+import gql from 'graphql-tag';
+import { useQuery } from 'react-apollo';
+
+const BenchmarkComponent = () => {
```
Comment (Raemon) [inline packages/lesswrong/components/BenchmarkComponent.jsx]: Is this from a separate branch? Seems sort of random?

Comment (jimrandomh) [reply packages/lesswrong/components/BenchmarkComponent.jsx]: Whoops, yes, that was a thing hanging around in my working directory that didn't belong in this PR

**Comment by Raemon** [comment]
On dev server, I notice that there (seem to be?) several instances of both ":decision theory" and "programming" tags, and not sure if that's just a testing artifact

image

File: packages/lesswrong/server/tagging/tagCallbacks.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  const normalizedName = normalizeTagName(tag.name);
+  if (tag.name !== normalizedName) {
+    tag = {
+      ...tag,
+      name: normalizedName,
+    };
+  }
+  
+  // Name must be unique
+  const existing = Tags.find({name: normalizedName, deleted:false}).fetch();
```
Comment (Raemon) [inline packages/lesswrong/server/tagging/tagCallbacks.js]: I'm somewhat surprised this infrastructure doesn't already exist in a more dry format. Not objecting to merging with it, just surprised.

File: packages/vulcan-core/lib/modules/containers/withCreate.js [hunk 3]
```diff
@@ -73,3 +73,27 @@ const withCreate = options => {
 };
 
 export default withCreate;
+
+// TODO: Only include in branch if actually used
+export const useCreate = ({
```
Comment (Raemon) [inline packages/vulcan-core/lib/modules/containers/withCreate.js]: Flagging that @Discordius should probably look over this part.

**Comment by Raemon** [comment]
(also note that I made a couple more UI tweaks as I reviewed the branch)

**Pull request merged**